File: app/src/main/java/com/robinson/notewithreminder/utilities/StopReminderReceiver.java
package com.robinson.notewithreminder.utilities;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import androidx.core.app.NotificationManagerCompat;
import java.util.Objects;

public class StopReminderReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Objects.requireNonNull(intent.getAction())
                .equals("StopReminderReceiver")) {

            new MusicControl(context).stopMusic();
            long id = intent.getLongExtra(CommonParameters.noteId, 0);
            NotificationManagerCompat.from(context).cancel((int) id);
        }
    }
}

File: app/src/main/java/com/robinson/notewithreminder/utilities/StopReminderReceiver.kt
package com.robinson.notewithreminder.utilities

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.core.app.NotificationManagerCompat
import java.util.Objects

class StopReminderReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (Objects.requireNonNull(intent.action)
            == "StopReminderReceiver"
        ) {
            MusicControl.getInstance(context)!!.stopMusic()
            val id = intent.getLongExtra(CommonParameters.noteId, 0)
            NotificationManagerCompat.from(context).cancel(id.toInt())
        }
    }
}